### PR TITLE
soroban-rpc: simulateTransaction: Get budget settings from the network

### PR DESCRIPTION
**Pull request by 2opremio**

### CODE REVIEW

File: cmd/soroban-rpc/lib/preflight/src/lib.rs [hunk 1]
```diff
@@ -107,7 +105,8 @@ fn preflight_invoke_hf_op_or_maybe_panic(
         golang_handle: handle,
     });
     let storage = Storage::with_recording_footprint(src);
-    let budget = Budget::default();
+    let compute = get_contract_compute_setting(&src)?;
+    let budget =
```
Comment (tsachiherman) [inline cmd/soroban-rpc/lib/preflight/src/lib.rs:109]: am I missing something here ?

**Comment by 2opremio** [comment]
Thanks for the review @dmkozh !

**Comment by 2opremio** [comment]
@tsachiherman PTAL

**Review by tsachiherman — APPROVED** [review]
looks great.

**Pull request merged**